pulsar-client/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -423,6 +427,23 @@ public ProducerConfiguration setInitialSequenceId(long initialSequenceId) {
         return this;
     }
 
+    /**
+     * Set optional key/value metadata with this producer.
+     * @param key
+     * @param value
+     * @return
+     */
+    public ProducerConfiguration setMetadata(String key, String value) {
+        checkArgument(key != null);
+        checkArgument(value != null);
+        metadata.put(key, value);
+        return this;
+    }
+
+    public Map<String, String> getMetadata() {
```

Comment:
there is `getMetadata()` that returns a map and `setMetadata()` that only sets 1 single value. 

For that I'd like more the `setProperty()` and `getProperties()`